neo/io/blackrockio.py

```diff
@@ -1816,43 +1817,54 @@ def __read_analogsignal(
         description = \
             "AnalogSignal from channel: {0}, label: {1}, nsx: {2}".format(
                 channel_idx, labels[idx_ch], nsx_nb)
-
-        data_times = np.arange(
-            t_start.item(), t_stop.item(),
-            self.__nsx_basic_header[nsx_nb]['period']) * t_start.units
-        mask = (data_times >= n_start) & (data_times < n_stop)
-        data_times = data_times[mask].astype(float)
-
-        sig_ch = signal[dbl_idx][:, idx_ch][mask].astype(float)
-
-        # transform dig value to pysical value
-        sym_ana = (max_ana[idx_ch] == -min_ana[idx_ch])
-        sym_dig = (max_dig[idx_ch] == -min_dig[idx_ch])
-        if sym_ana and sym_dig:
-            sig_ch *= float(max_ana[idx_ch]) / float(max_dig[idx_ch])
+        
+        if lazy:
+            period = self.__nsx_basic_header[nsx_nb]['period']
+            length = int(((n_stop.rescale(nsx_time_unit)-n_start.rescale(nsx_time_unit))/\
+                            self.__nsx_basic_header[nsx_nb]['period']).magnitude)
+            lazy_shape = (length, )
+            sig_ch = []
```

Comment:
Not really important, but maybe make an array np.array([]) for better consistency/legibility with the else part.